index.js

```diff
@@ -69,14 +70,15 @@ module.exports = {
                 return all;
             }.bind(this), [])
         ).then(function(contexts) {
-            return when.reduce(contexts, function(prev, context) {
+            return when.reduce(contexts, function(prev, context, idx) {
+                portsStarted.push(context); // collect ports that are started
```

Comment:
There is still no guarantee that the port had been started. This part of the code could be just:
return context.start().then(() => portsStarted.push(context));